bootspec/src/v1.rs

```diff
@@ -100,8 +103,14 @@ impl GenerationV1 {
 
         let init = generation.join("init");
 
-        let initrd = fs::canonicalize(generation.join("initrd"))
-            .map_err(|e| format!("Failed to canonicalize the initrd:\n{}", e))?;
+        let initrd = if generation.join("initrd").exists() {
+            Some(
+                fs::canonicalize(generation.join("initrd"))
```

Comment:
I would like this better if we didn't calculate the initrd path twice ;)

```suggestion
        let initrd_path = generation.join("initrd");
        let initrd = if initrd_path.exists() {
            Some(
                fs::canonicalize(initrd_path)
```